CRM/Event/BAO/Participant.php

```diff
@@ -1419,6 +1418,11 @@ public static function sendTransitionParticipantMail(
             'isConfirm' => $mailType === 'Confirm',
             'checksumValue' => $checksumValue,
           ],
+          'modelProps' => [
+            'participantID' => (int) $participantId,
```

Comment:
@demeritcowboy all those blanks rely on this being set here - that's where the token setting comes from 

![image](https://github.com/civicrm/civicrm-core/assets/336308/c992c1de-f1a0-4d38-9f17-b83af0c5004e)